### PR TITLE
System containers

### PR DESCRIPTION
**Pull request by giuseppe**

### CODE REVIEW

File: container.fc [hunk 1]
```diff
@@ -24,6 +25,8 @@
 /var/lib/docker/overlay(/.*)?	gen_context(system_u:object_r:container_share_t,s0)
 /var/lib/docker/overlay2(/.*)?	gen_context(system_u:object_r:container_share_t,s0)
 
+/var/lib/containers/atomic(/.*)?	gen_context(system_u:object_r:svirt_sandbox_file_t,s0)
```
Comment (rhatdan) [inline container.fc]: THis should be container_share_t.  This is a label that a container process can read/execute but can not write.

svirt_sandbox_file_t has been changed to container_file_t, but this label is writable by all container processes if the MCS label is s0.

**Comment by giuseppe** [comment]
nice, I tested with container_share_t and it works.  I have pushed a new version here ⬆️

**Comment by giuseppe** [comment]
there is a change that needs this in atomic.  How could we ensure that this path exists and it has the correct label?  Should I invoke "restorecon" from atomic if it doesn't exist and it is created?

**Comment by rhatdan** [comment]
No we need to handle this automatically. LGTM

**Comment by giuseppe** [comment]
a problem I have found now is that using container_share_t for /var/lib/containers/atomic then the container process has the label spc_t, I think because of:
```domtrans_pattern(container_runtime_t, container_share_t, spc_t)```

If I try to run a non privileged container ```docker run -ti --rm busybox echo hi```, then I hit this issue:

Dec 12 11:44:42 atomic-0011.localdomain kernel: type=1400 audit(1481543082.254:614): avc:  denied  { transition } for  pid=33714 comm="exe" path="/bin/echo" dev="dm-8" ino=8390657 scontext=system_u:system_r:spc_t:s0 tcontext=system_u:system_r:svirt_lxc_net_t:s0:c180,c678 tclass=process

**Comment by rhatdan** [comment]
Didn't you specify the docker process to run as docker_t?  Is containerd and runc running as spc_t?

ps -eZ | grep spc_t

I can add that transition rule.

**Comment by rhatdan** [comment]
Looks like the policy allows this in Fedora.

```
# cat > /tmp/t
Dec 12 11:44:42 atomic-0011.localdomain kernel: type=1400 audit(1481543082.254:614): avc: denied { transition } for pid=33714 comm="exe" path="/bin/echo" dev="dm-8" ino=8390657 scontext=system_u:system_r:spc_t:s0 tcontext=system_u:system_r:svirt_lxc_net_t:s0:c180,c678 tclass=process
sh-4.3# audit2allow  -i /tmp/t


#============= spc_t ==============

#!!!! This avc is allowed in the current policy
allow spc_t svirt_lxc_net_t:process transition;
```

**Comment by giuseppe** [comment]
It is running like this:
```
system_u:system_r:container_runtime_t:s0 33940 ? Ssl   0:00 /bin/runc start docker
system_u:system_r:spc_t:s0       33958 ?        Ssl    0:36 /usr/bin/docker-current daemon ....
```

**Comment by giuseppe** [comment]
if I add a module with "allow spc_t svirt_lxc_net_t:process transition;" then the container works:

# docker run -ti --rm busybox echo hi
hi

**Comment by rhatdan** [comment]
Yes I see this rule in current container-selinux.  So we should be all set with RHEL 7.3.2

**Comment by rhatdan** [comment]
Only problem with this is we would rather not run docker as spc_t, since this would allow all domains to communicate with it.  Can we specify in the runc config to run as the current label?

**Comment by giuseppe** [comment]
if I understand it correctly we have a transition to "spc_t" when the runc (container_runtime_t) executes docker (container_share_t).  At this point it can't change its label back to "container_runtime_t".  I've tried to set "container_runtime_t" in the config.json "selinuxLabel" but it doesn't work

**Comment by rhatdan** [comment]
What error did you see and what label did you run with.

system_u:system_r:container_runtime_t:s0

This should work?

**Comment by giuseppe** [comment]
if I understand it correctly we have a transition to "spc_t" when the runc (container_runtime_t) executes docker (container_share_t).  At this point it can't change its label back to "container_runtime_t".  I've tried to set "container_runtime_t" in the config.json "selinuxLabel" but it doesn't work.

I've tested with the last version of container-selinux with this patch:

```
commit b97d6cae2482d8fbf02e71a3cdd64bbc8950a134
Author: Dan Walsh <dwalsh@redhat.com>
Date:   Mon Dec 12 09:06:06 2016 -0500

    Allow contianer_runtimes to exec themselves
```

but it still doesn't solve the issue I am seeing, as the transition happens anyway.

Since we started tagging `container_share_t`everything under /var/lib/containers/atomic/* we will probably need some freedom later in the kind of types we can use.  Should we introduce a new type here that does not cause the `domtrans_pattern(container_runtime_t, container_share_t, spc_t)` transition?

**Comment by rhatdan** [comment]
No we have to keep that.  The only problem I see is getting the processes listening at /var/run/docker.socket to be labeled container_runtime_t.

Do you have a test machine I can reach to look at what is going on?

File: container.fc (hunk 1 above)
Comment (sushanco) [reply container.fc]: On which version was this change from 'svirt_sandbox_file_t' to 'container_file_t' introduced?

Comment (rhatdan) [reply container.fc]: I think it was introduced in the RHEL7.5 release.

**Pull request merged**